graylog2-web-interface/src/views/components/aggregationbuilder/HeatmapVisualizationConfiguration.tsx

```diff
@@ -0,0 +1,143 @@
+
+import Select from '../Select';
+
+const StyledInput = styled(Input)`
+  > label {
+    font-weight: normal;
+  }
+`;
+
+type Props = {
+  onChange: (config: HeatmapVisualizationConfig) => void,
+  config: HeatmapVisualizationConfig,
+};
+
+const _makeOption = (value) => ({ label: value, value });
+const colorScalesOptions = COLORSCALES.map(_makeOption);
+
+type Error = {
+  zmin?: string,
+  zmax?: string,
+  defaultValue?: string,
+}
+
+const _validateConfig = (config: HeatmapVisualizationConfig, setErrors): boolean => {
+  const { zMax, zMin, defaultValue } = config;
+  const errors = {} as Error;
+
+  if (zMin || zMax) {
+    if (zMin >= zMax) {
+      errors.zmin = 'Min is bigger than Max';
+    }
+
+    if (zMax <= zMin) {
+      errors.zmin = 'Max is smaller than Min';
+    }
+
+    if (defaultValue) {
+      if (defaultValue > zMax || defaultValue < zMin) {
+        errors.defaultValue = 'Default Value is out of range from Min and Max';
+      }
+    }
+  }
+
+  setErrors(errors);
+
+  return Object.keys(errors).length <= 0;
+};
+
+const _parseEvent = (event) => {
+  const { value } = event.target;
+
+  if (value === undefined) {
+    return undefined;
+  }
+
+  return parseFloat(value);
+};
+
+const HeatmapVisualizationConfiguration = ({ config = HeatmapVisualizationConfig.empty(), onChange: onChangeFunc }: Props) => {
+  const [errors, setErrors] = useState<Error>({});
+  const onChange = useCallback((newConfig) => {
+    if (_validateConfig(newConfig, setErrors)) {
+      onChangeFunc(newConfig);
+    }
+  }, [onChangeFunc]);
+  const modifyConfig = useCallback((fn: (HeatmapConfigBuilder) => HeatmapConfigBuilder) => {
+    onChange(fn(config.toBuilder()).build());
+  }, [config, onChange]);
+
+  const _onColorScaleChange = useCallback(({ value }) => modifyConfig((builder: HeatmapConfigBuilder) => builder.colorScale(value)), [modifyConfig]);
+  const _onReverseScaleChange = useCallback((e) => modifyConfig((builder: HeatmapConfigBuilder) => builder.reverseScale(e.target.checked)), [modifyConfig]);
+  const _onAutoScaleChange = useCallback((e) => modifyConfig((builder: HeatmapConfigBuilder) => builder.autoScale(e.target.checked)), [modifyConfig]);
+  const _onZminChange = useCallback((e) => modifyConfig((builder: HeatmapConfigBuilder) => builder.zMin(_parseEvent(e))), [modifyConfig]);
+  const _onZmaxChange = useCallback((e) => modifyConfig((builder: HeatmapConfigBuilder) => builder.zMax(_parseEvent(e))), [modifyConfig]);
+  const _onUseSmallestAsDefaultChange = useCallback((e) => modifyConfig((builder: HeatmapConfigBuilder) => builder.useSmallestAsDefault(e.target.checked)), [modifyConfig]);
+  const _onDefaultValueChange = useCallback((e) => modifyConfig((builder: HeatmapConfigBuilder) => builder.defaultValue(_parseEvent(e))), [modifyConfig]);
+
+  return (
+    <>
+      <span>Color Scheme</span>
+      <Select placeholder="Select Color Scheme"
+              onChange={_onColorScaleChange}
+              options={colorScalesOptions}
```

Comment:
Can we add `isClearable={false}` here? IMO it does not make sense to show the clear icon in here.